### PR TITLE
Use the octavia network.json network for Octavia (SOC-10904)

### PR DESCRIPTION
**Pull request by stefannica**
(based on #3827 )

Updates the automation scripts for Octavia, to add support
for configuring a network.json octavia network instead of the
previous way of configuring the Octavia management network
as an externally managed provider network.
 * to aid with debugging, intermediate network.json steps are backed
up so that it's easier to debug the myriad of `sed` commands that
modify `network.json` before bootstrapping crowbar
 * the Neutron barclamp calculates the `vlan_min` value for the ml2
plugin based on the fixed network VLAN ID. In order to avoid VLAN ID
collisions this change moves the fixed network VLAN ID to be above
that used for `octavia` and other VLAN networks
 * adds a `want_octavia_network` option that can be used to switch
between using the octavia network in network.json (default) and
the previous approach of relying on an externally configured provider
network
 * remove the `manage_cidr` attribute from the generated octavia
baclamp. This attribute has been removed and the Octavia recipes
rely entirely on the information extracted from network.json to configure the
management network

### CODE REVIEW

**Comment by stefannica** [comment]
I just realized that the octavia network is always set up, even if it's deleted from network.json and octavia isn't deployed yet. I need to address this somehow.

**Comment by stefannica** [comment]
Cloud/automation/blob/94e3a09d646acd67e6f41b54df0399690c553112/scripts/qa_crowbarsetup.sh#L4194-L4221) still works: https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/892/pipeline/
- cloud deployment excluding Octavia, then manually deploying Octavia with the use of an `octavia` network (i.e. validates that a customer can deploy Octavia with an `octavia` supplied neutron provider network on top of an existing cloud installation): https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/881/pipeline/
- complete cloud deployment, including Octavia, starting with the `octavia` network added to network.json:
  - using the OVS backend: https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/887/pipeline/
  - using the linuxbridge backend: https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/888/pipeline/

All tests were executed with the following patches applied:
 - the PR removing the `octavia` network from network.json: https://github.com/crowbar/crowbar-core/pull/2023
 - the PR reworking the octavia management network implementation around the `octavia` network.json network: https://github.com/crowbar/crowbar-openstack/pull/2385 (this PR)
 - the automation PR adding support for the `octavia` network.json network: https://github.com/SUSE-Cloud/automation/pull/3856

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -6424,12 +6466,19 @@ function onadmin_batch
...
+            get_novacontroller
+            setup_octavia_cert /etc/octavia/certs
+        fi
     fi
 
     safely crowbar batch $exclude --timeout 3600 build ${scenario}
-    if grep -q "barclamp: octavia" ${scenario}; then
+    if grep -q "barclamp: octavia" ${scenario} && [[ $want_octavia_network != 1 ]]; then
         get_novacontroller
         setup_octavia_cert /etc/octavia/certs
```
Comment (guangyee) [inline scripts/qa_crowbarsetup.sh:6483]: is this done in line 6475 already?

Comment (stefannica) [reply scripts/qa_crowbarsetup.sh:6483]: The opposite, actually. The story behind this silly looking condition is we have two situations:
 - either we add the `octavia` network to network json (`want_octavia_network=1`) and the Octavia barclamp creates the `lb-mgmt-net` network automatically. In this case, we can deploy all the barclamps in one shot.
 - or we don't (`want_octavia_network=0`), in which case we must create the `lb-mgmt-net` network manually (i.e. call the function at line 4194) before we deploy Octavia, but after we deploy Neutron. In this case, we have to deploy everything _except Octavia_, then create the network, then deploy Octavia by itself

This is to ensure we have logic in the automation scripts to cover both scenarios: that in which the customer wants to use the recommended automated way of configuring the `lb-mgmt-net` network by relying on the information supplied in network.json, the other in which customers will provide their own custom `lb-mgmt-net` network

**Pull request merged**